### PR TITLE
VTMUNC Authentication

### PR DESCRIPTION
**Pull request by quinnandersonvt**
Uses Jose library, supports basic authentication as the correct user email and password will create a cookie in which the user can navigate to an admin page.

### CODE REVIEW

**Comment by erland-syafiq** [comment]
Overall, looks good, just some issues here and there, but excellent progress!

File: site/middleware.js [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+            } catch (error) {
+                // If any decryption errors then redirect to login
+                return NextResponse.redirect(loginUrl);
+            }
+        } else {
+            return NextResponse.redirect(loginUrl);
+        }
+    }
     
-    if (/[A-Z]/.test(request.nextUrl.pathName)) {
```
[the remark this answers is not in the document]
Comment (quinnandersonvt) [reply site/middleware.js:5]: This was here before, I can remove though

File: site/app/login/LoginForm.jsx [hunk 2]
```diff
@@ -22,6 +22,15 @@ export default function LoginForm() {
...
             // Attempt login
+            const userEmail = formData.email;
+            const userPass = formData.password;
+            // Send post request with user email and password
+            fetch('/api/auth/login/', {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/json' },
+                body: JSON.stringify({ userEmail, userPass }),
+            });
+
```
[the remark this answers is not in the document]
Comment (quinnandersonvt) [reply site/app/login/LoginForm.jsx:45]: Added it now, I did it based on response which is ok (though I do not know if it is best 'coding practice'.

[the remark this answers is not in the document]
Comment (erland-syafiq) [reply site/app/login/LoginForm.jsx:45]: It's probably fine, I think? I don't recall how we did it in systems.

**Comment by erland-syafiq** [comment]
Looks good

**Pull request merged**